scripts/analyse-kernel.sh

```diff
@@ -0,0 +1,156 @@
+	echo "  --configfile, analysisconfig file for build"
+	echo "Example: $0 -r linux-stable -s d8a5b80568a9 -c defconfig -i [absolute-path-to-file]"
+	echo "Example with analysisconfig file: $0 --configfile [absolute-path-to-file]"
+	echo "Before use it, please be sure you set KERNEL_SRC_BASE variable correctly"
+	exit 1
+}
+set_kernel_repository() {
+	if [ ! -z "$1" ]; then
+		case $1 in
+			torvalds | stable | next )
+			if [ "$1" == "torvalds" ]; then
+				KERNEL_REPOSITORY="$KERNEL_SRC_BASE/torvalds/linux"
+			elif [ "$1" == "stable" ]; then
+				KERNEL_REPOSITORY="$KERNEL_SRC_BASE/stable/linux-stable"
+			elif [ "$1" == "next" ]; then
+				KERNEL_REPOSITORY="$KERNEL_SRC_BASE/next/linux-next"
+			else
+				echo "Invalid Kernel Repository Parameter!"
+				echo "You can pass -> torvalds | stable | next"
+				exit 1;
+			fi
+		esac
+	else
+		echo "You provided an empty KERNEL_REPOSITORY parameter!"
+		echo "Acceptable parameters are =  torvalds | stable | next"
+		exit 1;
+	fi
+}
+set_kernel_config() {
+	if [ ! -z "$1" ]; then
+		case $1 in
+			allnoconfig | allmodconfig | allyesconfig | defconfig | randconfig )
+			KERNEL_CONFIG="$1"
+			;;
+	*)
+		echo "You provided a KERNEL_CONFIG parameter but it is not valid!"
+		echo "Valid Parameters are = allnoconfig | allmodconfig | allyesconfig | defconfig | randconfig"
+		exit 1;
+	esac
+	else
+		echo "You provided an empty KERNEL_CONFIG parameter" ##Can it happen in practice??
+		echo "Valid Parameters are = allnoconfig | allmodconfig | allyesconfig | defconfig | randconfig"
+		exit 1;
+	fi
+}
+set_inferconfig_file_location() {
+	if [ -f "$SCRIPTS_DIRECTORY/$1" ]; then
+		INFERCONFIG_LOCATION="$SCRIPTS_DIRECTORY/$1"
+	elif [ -f "$1" ]; then
+		INFERCONFIG_LOCATION="$1"
+	fi
+}
+set_scripts_directory() {
+	SCRIPTS_DIRECTORY="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+}
+read_and_set_variables_from_analysisconfig() {
+	if [ -f "$1" ]; then
+		source $1
+	else
+		echo "Couldn't read analysisconfig file!"
+		exit 1;
+	fi
+}
+check_kernel_src_base_valid() {
+	if [ -z "$KERNEL_SRC_BASE" ]; then 
+		echo "Please set KERNEL_SRC_BASE first"
+		help
+		exit 1
+	elif [ ! -d "$KERNEL_SRC_BASE" ]; then
+		echo "KERNEL_SRC_BASE does not point to a directory"
+		help
+		exit 1
+	fi
+}
+check_kernel_repository_valid() {
+	if [ -z "$KERNEL_REPOSITORY" ]; then
+		echo "You must provide a target kernel repository"
+		echo "Valid parameters are =  torvalds | stable | next"
+		exit 1;
+	elif [ ! -d "$KERNEL_REPOSITORY" ]; then # Check KERNEL_REPOSITORY is a directory or not
+		echo "Path $KERNEL_REPOSITORY doesn't point to a directory"
+		echo "Please fix it and then run script"
+		exit 1
+	fi
+}
+check_kernel_configuration_valid() {
+	if [ -z "$KERNEL_CONFIG" ]; then # Check KERNEL_CONFIG variable is set
+		echo "You must provide a valid kernel configuration"
+		echo "Valid Parameters are = allnoconfig | allmodconfig | allyesconfig | defconfig | randconfig"
+		exit 1;
+	fi
+}
+check_inferconfig_exists() {
```

Comment:
infer --help shows out some environment variables 

```
ENVIRONMENT
       Extra arguments may be passed to all infer commands using the
       INFER_ARGS environment variable (see the OPTIONS section). INFER_ARGS
       is expected to contain a string of ^-separated options. For instance,
       calling `INFER_ARGS=--debug^--print-logs infer` is equivalent to
       calling `infer --debug --print-logs`.

       INFERCONFIG: Tells infer where to find the .inferconfig file. (See the
       FILES section)

       If INFER_STRICT_MODE is set to "1", then infer commands will exit with
       an error code in some cases when otherwise a simple warning would be
       emitted on stderr, for instance if a deprecated form of an option is
       used.

```
 Should they be used?

Reply:
That is certainly a good idea.

Reply:
Thanks for reviewing @phongt ,
And this a definitely good idea. I will use this feature in the script, and then update pull request